pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go

```diff
@@ -2,30 +2,186 @@ package targetconfigcontroller
+var kscNo string = string(v410_00_assets.MustAsset("v4.1.0/config/defaultconfig-postbootstrap-noscoring.yaml"))
+
+var cmLow = &corev1.ConfigMap{
+	TypeMeta: metav1.TypeMeta{APIVersion: "v1", Kind: "ConfigMap"},
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "config",
+		Namespace: "openshift-kube-scheduler",
+	},
+	Data: map[string]string{"config.yaml": kscLow},
+}
+
+var cmHigh = &corev1.ConfigMap{
+	TypeMeta: metav1.TypeMeta{APIVersion: "v1", Kind: "ConfigMap"},
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "config",
+		Namespace: "openshift-kube-scheduler",
+	},
+	Data: map[string]string{"config.yaml": kscHigh},
+}
+
+var cmNo = &corev1.ConfigMap{
+	TypeMeta: metav1.TypeMeta{APIVersion: "v1", Kind: "ConfigMap"},
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "config",
+		Namespace: "openshift-kube-scheduler",
+	},
+	Data: map[string]string{"config.yaml": kscNo},
+}
+
+func Test_manageKubeSchedulerConfigMap_v311_00_to_latest(t *testing.T) {
+
+	fakeRecorder := NewFakeRecorder(1024)
+
+	type args struct {
+		client                corev1client.ConfigMapsGetter
+		recorder              events.Recorder
+		configSchedulerLister configlistersv1.SchedulerLister
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *corev1.ConfigMap
+		want1   bool
+		wantErr bool
+	}{
+		{
+			name: "unknown-cluster",
+			args: args{
+				recorder: fakeRecorder,
+				configSchedulerLister: &fakeSchedConfigLister{
+					Items: map[string]*configv1.Scheduler{"unknown": configLow},
+				},
+			},
+			want:    nil,
+			want1:   false,
+			wantErr: true,
+		},
+		{
+			name: "unknown-profile",
+			args: args{
+				recorder: fakeRecorder,
+				configSchedulerLister: &fakeSchedConfigLister{
+					Items: map[string]*configv1.Scheduler{"cluster": configUnknown},
+				},
+			},
+			want:    nil,
+			want1:   false,
+			wantErr: true,
+		},
+		{
+			name: "low-node-utilization",
+			args: args{
+				recorder: fakeRecorder,
+				configSchedulerLister: &fakeSchedConfigLister{
+					Items: map[string]*configv1.Scheduler{"cluster": configLow},
+				},
+			},
+			want:    cmLow,
+			want1:   true,
+			wantErr: false,
+		},
+		{
+			name: "high-node-utilization",
+			args: args{
+				recorder: fakeRecorder,
+				configSchedulerLister: &fakeSchedConfigLister{
+					Items: map[string]*configv1.Scheduler{"cluster": configHigh},
+				},
+			},
+			want:    cmHigh,
+			want1:   true,
+			wantErr: false,
+		},
+		{
+			name: "no-scoring",
+			args: args{
+				recorder: fakeRecorder,
+				configSchedulerLister: &fakeSchedConfigLister{
+					Items: map[string]*configv1.Scheduler{"cluster": configNo},
+				},
+			},
+			want:    cmNo,
+			want1:   true,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// need a client for each test
+			tt.args.client = fake.NewSimpleClientset().CoreV1()
+			got, got1, err := manageKubeSchedulerConfigMap_v311_00_to_latest(tt.args.client, tt.args.recorder, tt.args.configSchedulerLister)
```

Comment:
Given `tt.args.client` is used only once in this block, you might pass `fake.NewSimpleClientset().CoreV1()` directly into `manageKubeSchedulerConfigMap_v311_00_to_latest`

Reply:
Addressed in commit 764f14d9c4a0402f6d17b2fa017d13257f6f2bff